src/app/Providers/components/ProvidersTable/OpenShift/__tests__/OpenShiftProvidersTable.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { createMemoryHistory } from 'history';
+import '@testing-library/jest-dom';
+import { Router } from 'react-router-dom';
+
+import { NetworkContextProvider } from '@app/common/context';
+import OpenShiftProvidersTable from '../OpenShiftProvidersTable';
+import { MOCK_PROVIDERS } from '@app/queries/mocks/providers.mock';
+
+describe('<OpenShiftProvidersTable />', () => {
+  const history = createMemoryHistory();
+  const props = {
+    providers: MOCK_PROVIDERS.openshift,
+  };
+
+  it('renders openshift table', () => {
+    render(
+      <NetworkContextProvider>
+        <Router history={history}>
+          <OpenShiftProvidersTable {...props} />
+        </Router>
+      </NetworkContextProvider>
+    );
+
+    expect(
+      screen.getByRole('grid', { name: /OpenShift Virtualization providers table/ })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('row', { name: /ocpv-1 https:\/\/my_OCPv_url 41 26 8 0 Ready Actions/ })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('row', { name: /ocpv-2 https:\/\/my_OCPv_url 41 26 8 0 Critical Actions/ })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('row', { name: /ocpv-3 https:\/\/my_OCPv_url 41 26 8 0 Ready Actions/ })
+    ).toBeInTheDocument();
+  });
+
+  // The expanding section doesn't render, is it the right button?
+  it.skip('renders storage classes', async () => {
+    render(
+      <NetworkContextProvider>
+        <Router history={history}>
+          <OpenShiftProvidersTable {...props} />
+        </Router>
+      </NetworkContextProvider>
+    );
+
+    const sharedClasses = screen.getAllByRole('button', { name: '0' });
+    userEvent.click(sharedClasses[2]);
```

Comment:
It would be nice to have the selectors be a little more semantic instead of hard-coding array indexes like this.. we can leave this for the PR that fixes this test, but it would be better to instead e.g. find the row that contains that provider name, traverse from there to find the button, etc. Traversing the document like a user would with their eyes will make the tests less brittle.

Reply:
That would be much better meanwhile once a row is queried, react testing libraries can't be used to query on HTMLElement.

Reply:
Interesting. Maybe there's a different better way then, we can figure that out later.